### DISCUSSION TITLE
FLE Import

### DISCUSSION THREAD

**Discussion by DK1MI** [magicbug/Cloudlog#2614] [Ideas / Feature Requests]
It would be great to have an import function that allows importing files in FLE ([Fast Log Entry](https://df3cb.com/fle/)) format. I myself like to log in Vim with a syntax highlighting plugin for the FLE format but then have to convert it first into ADIF before importing it into Cloudlog. 

SOTA hams seem to also use FLE so they have added a FLE import functionality (see [https://reflector.sota.org.uk/t/fast-log-entry-syntax-highlighting-for-vim/33645](https://reflector.sota.org.uk/t/fast-log-entry-syntax-highlighting-for-vim/33645)). This format seems to become more popular.

Thanks and 73
Michael, DK1MI

**Comment by HB9HIL** [comment]
We will take a look at that. :-)

**Comment by phl0** [comment]
Could you share the fle plugin for vim or point me to the source?

**Reply by HB9HIL** [reply]
This ?

https://reflector.sota.org.uk/t/fast-log-entry-syntax-highlighting-for-vim/33645

**Reply by DK1MI** [reply]
This is the plugin but it really doesn't do more than to just highlight keywords: https://dk1mi.radio/vim-fle-syntax/
Here is the original tool for converting logs in FLE syntax to ADIF: https://df3cb.com/fle/
And here is an implementation in go: https://github.com/on4kjm/FLEcli

Thanks for considering it!

**Reply by AndreasK79** [reply]
SFLE also exists https://github.com/ok2cqr/sfle

It is a web version.

**Reply by DK1MI** [reply]
Oh nice! I was not aware of this

**Reply by magicbug** [reply]
> SFLE also exists https://github.com/ok2cqr/sfle
> 
> It is a web version.

You beat me to it, its always been an item I planned on integrating but like everything one of many things.

**Reply by HB9HIL** [reply]
@magicbug I will try to implement that with the help of the rest of the dev guys 👍

**Reply by magicbug** [reply]
For those waiting DF3CB's FLE app is fantastic for post logging if you have been involved with anyone who paper logs

**Reply by DK1MI** [reply]
That would be fantastic! Thanks for considering the implementation

**Comment by AndreasK79** [comment]
@DK1MI the latest dev branch now contains the SimpleFLE code. @HB9HIL did a great job with integrating it. Please test and report back.

**Reply by DK1MI** [reply]
Excellent! Thank you very much for your work on this. Problem is that I am a simple subscriber and am enjoying the hosted solution so I fear I am not able to test the dev branch.

**Reply by HB9HIL** [reply]
You can wait for the next release 😎👍

**Reply by HB9HIL** [reply]
> Excellent! Thank you very much for your work on this. Problem is that I am a simple subscriber and am enjoying the hosted solution so I fear I am not able to test the dev branch.

@magicbug just released the new Version 2.5. So you will surely be able to test SimpleFLE 😊👍

**Comment by HB9HIL** [comment]
@DK1MI Were you able to test SimpleFLE? Could you give any feedback?

**Reply by DK1MI** [reply]
Sadly not. I cannot find the feature in the hosted version.

**Reply by HB9HIL** [reply]
Should be in QSO -> Simple Fast Log Entry

**Comment by pa3hfj** [comment]
Well, I saw the announcement and decided to do a test yesterday and I can say that the new functionality is really a nice addition to Cloudlog! It came in handy for the (long postponed :-) ) task of entering the hand-written logs of my XYL into her newly created logbook in my Cloudlog instance. It saves a lot of time, so thanks for the work on this.

I have just two remarks:

- When you put in the frequency of the QSO's the right band is chosen, but the frequency is not entered in the log, just a 0 is to be seen.
- The date stays fixed, also when a different date is entered in the inputbox. This means that if you have to type over a log with a lot of different dates (as was my case) it takes somewhat more work in the way of some mouse clicks for saving the QSO's and revert to the input task.

But overall, a very nice job!

**Reply by HB9HIL** [reply]
Hello PA3HFJ

First of all thank you very much for the feedback. This is important to make Cloudlog better and Users happier!
Could you provide my the text you typed in to test it? 

1) It seems like there is a bug but I have to figure out where the problem is. If I can get your text it's may be easier to debug. 

2) This is probably not a bug, more a syntax mistake. The command "date" needs its own line and have to be written in the format YYYY-MM-DD. If I test some data and add various dates it works without problems. Try again with that particular format. 

Vy 73 de HB9HIL

**Reply by HB9HIL** [reply]
for example `date 2023-04-23` for the 23th of April 2023

**Reply by HB9HIL** [reply]
if the new date is just a few days AFTER the latest one you can use `day +`

The number of `+` represents the number of days

For example 

date `2023-04-23` is the date 23.04.2023
then you type your qso's
then `day +++` results in the date 26.04.2023

**Reply by HB9HIL** [reply]
PS: something like `day -` is not implemented

**Reply by pa3hfj** [reply]
Thanks for your answer! A nice "feature" of an online logbook is that you can reach it from work ;-) , so here we go: 

1. This is a sample line: "1200 pa1test 3.666 54 56 ssb" I also tried putting the frequency and mode on a first seperate line but that makes no difference.
2. It was indeed a mistake. Because the datebox has the form of dd-mm-yyyy I probably used that format also in the inputbox. I tried it now the other way around yyyy-mm-dd and that works as expected! Maybe a small remark could be added to the Syntax Help window.

73, Wijnand PA3HFJ.

**Reply by HB9HIL** [reply]
Thanks for your reply. The QRG error will be fixed with the next stable release.

**Reply by HB9HIL** [reply]
@pa3hfj 

The new version was just released. The issues should be fixed now. :)

**Reply by pa3hfj** [reply]
Thank you Fabian, I can confirm the fix!
73, Wijnand PA3HFJ.

(PS: nice mobile shack on your QRZ page! I own a Land Rover Defender for the same purpose)

**Comment by UnicycleDumpTruck** [comment]
I love the Simple FLE, thank you for implementing it! I paper log for my POTA activations, and my usual workflow involves double checking the call signs in the QRZ callbook (to make sure the call is valid and that the state/location I copied matches the callsign's default) before or as I enter them into the fast log entry. Is there a straightforward way to have the Simple FLE in Cloudlog look up the entered callsign's location from the Callbook in the preview mode, so I know before adding the full list of QSOs if I have busted calls?

**Reply by magicbug** [reply]
I'm not sure I'd be relying on QRZ as the total knowledge of the truth but certainly could be looked into as an option
